wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/GroupDetailsViewController.swift

```diff
@@ -58,6 +65,11 @@ final class GroupDetailsViewController: UIViewController, ZMConversationObserver
                 }
             }
         }
+
+        Task { @MainActor in
+            await updateUserE2EICertificationStatuses()
+            collectionViewController.sections = computeVisibleSections()
+        }
```

Comment:
is this the same as :

```
Task {
            await updateUserE2EICertificationStatuses()
      let sections = computeVisibleSections()
       await MainActor.run {
            collectionViewController.sections = sections
       }
}
```

I moved the `computeVisibleSections` out because it's quite consuming and it *should* not be needed to be done on mainThread

Reply:
I don't think that would work that easily. That just brings us back to our ubiquitous issue of accessing core data instances from the wrong thread. I haven't tried it yet, but just from observing the code I expect it to run into issues as early as the filtering between admins and members happens:
```
let admins = participants.filter { $0.isGroupAdmin(in: conversation) }
```

Reply:
ok sorry maybe my point was two actually. You answered the second one about performance of `computeVisibleSections()`.

The first one is: Does this
``` 
Task { @MainActor in
            await updateUserE2EICertificationStatuses()
            collectionViewController.sections = computeVisibleSections()
}
```
equals
``` 
Task {
            await updateUserE2EICertificationStatuses()
            await MainActor.run {
               collectionViewController.sections = computeVisibleSections()
             }
}
```
I wonder if in the first case we're blocking mainthread somehow since we're waiting for the updateUserE2EICertificationStatuses to be finished?